.github/workflows/image-build.yml

```diff
@@ -0,0 +1,106 @@
+name: Build and push images
+
+on:
+  push:
+    branches-ignore: ['releases/**']
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+
+  build-and-push-images:
+
+    runs-on: ubuntu-latest
+
+    # Sets the permissions granted to the `GITHUB_TOKEN` for the actions in
+    # this job.
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+
+    outputs:
+      dir_changes: ${{ steps.directory-changes.outputs.changed_dirs }}
+
+    steps:
+      - name: Debug Event Payload
+        env:
+          EVENT_PAYLOAD: ${{ toJson(github.event) }}
+        run: echo "$EVENT_PAYLOAD"
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Git branch name
+        id: git-branch-name
+        run: echo "${{ github.head_ref || github.ref_name }}"
+      
+
+      - name: Directory Changes
```

Comment:
Your call but maybe we put this in the Python code, passing in the event name and base ref instead of inlining it here?  Rationale: it's not possible to test here.

Reply:
It is a copy-paste of the core part in the action that you used initially. Not sure how this can be tested. I will see.

Reply:
If you mean the gitlab stuff, yeah, but I purposely didn't put anything but the least amount of code in the yaml in the new github stuff.